Dark Mode Support

Basketballforce opened the pull request:
Hey there Dolphin team!

I figured it was time that the Dolphin site had a dark/light mode toggle so I took a stab at it.

These changes include:
-  Dark mode css
-  Dark/Light toggle that uses browser/os preferences, local storage and icon toggling to change
- Curated edits for a pleasant dark mode experience
- Made up of 1 main css file, small edits to the main css styling, and js logic to handle icon switching
- Css for every (known) page on the dolphin-emu domain

Notes:
 - Per the contributing instructions, the static and media file paths appear to be overwritten in Django. I'm not sure if this will occur but, I imagine it is possible that the added css and js href/script tag sources may be off in other people's environment. If so, let me know and I will take a look at fixing and providing screen shots so people can get an idea of what was changed

- I am able to test css edits on dolphin pages hosted publicly (using chrome dev tools) but, was unable to test some locally due to database limitations/Django db connections that appear to be outside my control.

Thank you!
BasketballForce

Basketballforce committed:
add dark mode css styling

Basketballforce committed:
add blank css sheet to revert dark mode changes

Basketballforce committed:
add icon styling for sun and moon / light and dark icons on navbar

Basketballforce committed:
Add js file to handling toggling dark/light mode and light/dark icon po
sition with mobile view and desktop view

Basketballforce committed:
Dark mode done! add required html ids for light/dark icons. Add light/sun icon. link cs
s and js file

Basketballforce commented:
Uploading some screenshots for reference. A local version with just the plain html, css, and js can be found [here](https://github.com/Basketballforce/DarkDolphin)
![1](https://user-images.githubusercontent.com/28318597/150723133-339ed343-5422-47a8-a034-f36621edb941.png)
![2](https://user-images.githubusercontent.com/28318597/150723258-a9cb87bc-b20a-4650-97cb-321aa251b007.png)
![3](https://user-images.githubusercontent.com/28318597/150723291-bd0a9714-e2a6-4c22-a6d3-c781761fe0fb.png)
![4](https://user-images.githubusercontent.com/28318597/150723298-5b3a1aea-38e2-4bfd-b6ef-111abab63953.png)
![5](https://user-images.githubusercontent.com/28318597/150723306-1d794f32-454f-497a-8d7f-08f5ad4239ab.png)
![6](https://user-images.githubusercontent.com/28318597/150723311-c4009bf5-ac9e-48da-af3a-30436905882c.png)
![7](https://user-images.githubusercontent.com/28318597/150723319-de8a7c77-ca4e-4a1a-9ebc-a4d12e3a3bc8.png)

MayImilae commented:
Why orange?

Basketballforce commented:
Its the color invert (inverted color) of the Dolphin blue logo.

 I think it gives the dark mode its own unique flare but, changing this back to the classic blue would be very easy and do-able if preferred!

MayImilae commented:
Definitely would be preferred. We use blue everywhere already, plus... bright orange on black is rather piercing imo.

Basketballforce commented:
I went and reverted the color changes from orange to the original blue. I have attached some screenshots for reference. I haven't updated the pull request since the original blue in dark mode doesn't have the best color contrast and dark mode experience (imo). 

I will try to select a more "dark friendly" blue color scheme/theme that aligns to the original design. After which I will add more screenshots for reference and update the pull requests

![1ogBlue](https://user-images.githubusercontent.com/28318597/151715080-9e6b3b82-739a-48cc-9acd-a2defe11365d.png)
![ogBlue2](https://user-images.githubusercontent.com/28318597/151715087-b2b6fd51-7af7-4aa5-880d-b7a72f1b02f8.png)
![ogBlue3](https://user-images.githubusercontent.com/28318597/151715090-b85404fc-aaf9-451e-acf9-85c51e85e8f4.png)
![ogBlue4](https://user-images.githubusercontent.com/28318597/151715091-d9760c03-a8dc-4677-a8e9-570823a94fdf.png)
![ogBlue5](https://user-images.githubusercontent.com/28318597/151715093-9574e68b-7c9b-4f82-8798-b2a6e28dd66d.png)
![ogBlue6](https://user-images.githubusercontent.com/28318597/151715097-b3cbc6c6-442c-495b-9e16-395304b8005a.png)

Basketballforce committed:
Revert colors back to original blue by removing orange style edits. Make links on darker backgrounds lighter for contrast. Remove btn-info styling

Basketballforce committed:
add margin left to hamburger button for mobile view. Separate it from the dark/light icons

Add spacing between the light/dark mode toggle icons and the hamburger mobile toggle. For mobile view specifically.

Basketballforce commented:
After second review, the original blue looked perfectly fine for dark mode. Smaller additional edits have been made for quality of life. These are:

1) Links on the darker background/black background have been made a lighter shade of blue for color contrast sake.
2) Spacing between the light/dark mode icons and the hamburger toggle for mobile view has been increased for clearer/consistent  separation of elements.
3) Button styling for the downloads page (btn-info) has been reverted back to original

The code has been updated/pushed to the pull request. Dark mode looks very similar to the most recent changes above in which dark mode was reverted to blue. I will add one or two screenshots below to reflect the new changes.

![updatedBlue](https://user-images.githubusercontent.com/28318597/152664625-8e0b64e9-9623-499e-b23d-8f6b3fb7dcd8.png)
![updatedBlue2](https://user-images.githubusercontent.com/28318597/152664627-ec1fd5f6-1eb0-4593-bd75-defe817f84e8.png)
![HamburgerSeperate](https://user-images.githubusercontent.com/28318597/152664629-6e529841-5586-4ce4-998d-9898c8dc8548.png)

osfanbuff63 commented:
This looks great, and I'm picky with dark mode :P 

For comparison, here's [Dark Reader](https://darkreader.org)'s automatic dark mode:
![Screenshot 2022-06-22 181505](https://user-images.githubusercontent.com/91388253/175162120-3f5fdb73-55a5-4135-9713-4425e72c820e.png)

The native dark mode is almost always better than that too.
